packages/e2e-tests/specs/editor/various/block-directory-add.test.js

```diff
@@ -0,0 +1,166 @@
+ */
+import { enableExperimentalFeatures } from '../../../experimental-features';
+
+// Urls to mock
+const SEARCH_URLS = [
+	'/__experimental/block-directory/search',
+	`rest_route=${ encodeURIComponent(
+		'/__experimental/block-directory/search'
+	) }`,
+];
+
+const INSTALL_URLS = [
+	'/__experimental/block-directory/install',
+	`rest_route=${ encodeURIComponent(
+		'/__experimental/block-directory/install'
+	) }`,
+];
+
+// Example Blocks
+const mockBlock1 = {
+	name: 'block-directory-test-block/main-block',
+	title: 'Block Directory Test Block',
+	description: 'This plugin is useful for the block.',
+	id: 'block-directory-test-block',
+	rating: 0,
+	rating_count: 0,
+	active_installs: 0,
+	author_block_rating: 0,
+	author_block_count: 1,
+	author: 'No Author',
+	icon: 'block-default',
+	assets: [
+		'fake_url.com/block.js', // we will mock this
+	],
+	humanized_updated: '5 months ago',
+};
+
+const mockBlock2 = {
+	...mockBlock1,
+	name: 'block-directory-test-block/secondary-block',
+	title: 'Block Directory Test Block - Pt Deux',
+	id: 'block-directory-test-secondary-block',
+};
+
+// Block that will be registered
+const block = `( function() {
+	var registerBlockType = wp.blocks.registerBlockType;
+	var el = wp.element.createElement;
+
+	registerBlockType( '${ mockBlock1.name }', {
+		title: 'Test Block for Block Directory',
+		icon: 'hammer',
+		category: 'common',
+		attributes: {},
+		edit: function( props ) {
+			return el( 'p', null, 'Test Copy' );
+		},
+		save: function() {
+			return null;
+		},
+	} );
+} )();`;
+
+function getResponseObject( obj, contentType ) {
+	return {
+		status: 200,
+		contentType,
+		body: obj,
+	};
+}
+
+function createResponse( mockResponse, contentType = undefined ) {
+	return async ( request ) =>
+		request.respond( getResponseObject( mockResponse, contentType ) );
+}
+
+const matchUrl = ( reqUrl, urls ) => {
+	return urls.some( ( el ) => reqUrl.indexOf( el ) >= 0 );
+};
+
+describe( 'adding blocks from block directory', () => {
+	beforeEach( async () => {
+		await enableExperimentalFeatures( [ '#gutenberg-block-directory' ] );
+		await createNewPost();
+	} );
+
+	it( 'Should show an empty state when no plugin is found.', async () => {
+		// Be super weird so there won't be a matching block installed
+		const impossibleBlockName = '@#$@@Dsdsdfw2#$@';
+
+		// Return an empty list of plugins
+		await setUpResponseMocking( [
+			{
+				match: ( request ) => matchUrl( request.url(), SEARCH_URLS ),
+				onRequestMatch: createResponse( JSON.stringify( [] ) ),
+			},
+		] );
+
+		// Search for the block via the inserter
+		await searchForBlock( impossibleBlockName );
+
+		const selectorContent = await page.evaluate(
+			() =>
+				document.querySelector( '.block-editor-inserter__block-list' )
+					.innerHTML
+		);
+		expect( selectorContent ).toContain( 'has-no-results' );
+	} );
+
+	it( 'Should be able to add (the first) block.', async () => {
+		// Setup our mocks
+		await setUpResponseMocking( [
+			{
```

Comment:
I was looking at other tests, and the [embedding tests](https://github.com/WordPress/gutenberg/blob/54c1b30cfe36e34edba521017c65036a661aac48/packages/e2e-tests/specs/editor/various/embedding.test.js#L85) set these responses as a constant before getting into the tests themselves — IMO it's easier to read that way, what do you think about making that change? Setting up maybe `MOCK_EMPTY_RESPONSES` & `MOCK_BLOCKS_RESPONSES` or something like that.